cmds/switch_root/utils.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+const AT_REMOVEDIR = 0x200
+
+// Code in this file comes from the syscall package.
+// Either the things are not out yet (fstatat) or private (unlinkat with flags)
+//
+// ulinkat because it's private
+// fstatat because it's commited but not released yet
+
+// Do the interface allocations only once for common
+// Errno values.
+var (
+	errEAGAIN error = syscall.EAGAIN
+	errEINVAL error = syscall.EINVAL
+	errENOENT error = syscall.ENOENT
+)
+
+// TODO remove when available/public in go runtime
+// errnoErr returns common boxed Errno values, to prevent
+// allocations at runtime.
+func errnoErr(e syscall.Errno) error {
+
+	switch e {
+	case 0:
+		return nil
+	case syscall.EAGAIN:
+		return errEAGAIN
+	case syscall.EINVAL:
+		return errEINVAL
+	case syscall.ENOENT:
+		return errENOENT
+	}
+	return e
+}
+
+// TODO remove when available/public in go runtime
```

Comment:
unlinkat will never make it into the syscall package. The syscall package is frozen in favor of https://godoc.org/golang.org/x/sys/unix, which contains Unlinkat and Fstatat. You should use that.

Reply:
Thanks for the pointer, I was wondering what the story with the syscall package was. Will migrate over.